#!/usr/bin/env python

PACKAGE = "lidar_team_erp42"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("minPoints", int_t, 0, "minPoints", 10, 1, 100)
gen.add("epsilon", double_t, 0, "epsilon", 0.4, 0, 20)
gen.add("minClusterSize", int_t, 0, "minClusterSize", 10, 1, 100)
gen.add("maxClusterSize", double_t, 0, "maxClusterSize", 1000, 1, 10000)

gen.add("xMinROI", double_t, 0, "xMinROI", -0.3, -200, 10)
gen.add("xMaxROI", double_t, 0, "xMaxROI", 8.0, -10, 200)
gen.add("yMinROI", double_t, 0, "yMinROI", -4.0, -200, 10)
gen.add("yMaxROI", double_t, 0, "yMaxROI", 4.0, -10, 200)
gen.add("zMinROI", double_t, 0, "zMinROI", -0.6, -2, 0)
gen.add("zMaxROI", double_t, 0, "zMaxROI", 2.0, -2, 100)

gen.add("xMinBoundingBox", double_t, 0, "xMinBoundingBox", 0.05, 0, 10);
gen.add("xMaxBoundingBox", double_t, 0, "xMaxBoundingBox", 2.0, 0, 10);
gen.add("yMinBoundingBox", double_t, 0, "yMinBoundingBox", 0.05, 0, 10);
gen.add("yMaxBoundingBox", double_t, 0, "yMaxBoundingBox", 2.0, 0, 10);
gen.add("zMinBoundingBox", double_t, 0, "zMinBoundingBox", 0.05, 0, 0.1);
gen.add("zMaxBoundingBox", double_t, 0, "zMaxBoundingBox", 2.0, 0, 10);

exit(gen.generate(PACKAGE, "track_waypoint", "hyper_parameter"))
